code/game/objects/structures/traps.dm

```diff
@@ -19,56 +33,43 @@
 		trap_effect(L)
 		animate(src, alpha = initial(alpha), time = time_between_triggers)
 
-/obj/structure/trap/examine(mob/user)
-	..()
-	if(!isliving(user)) //bad ghosts, stop trying to powergame from beyond the grave
-		return
-	user << "You reveal a trap!"
-	alpha = 200
-	animate(src, alpha = initial(alpha), time = time_between_triggers)
-
-
 /obj/structure/trap/proc/trap_effect(mob/living/L)
 	return
 
 /obj/structure/trap/stun
 	name = "shock trap"
-	desc = "A trap that will shock you, it will burn your flesh and render you immobile, You'd better avoid it."
+	desc = "A trap that will shock you and it will render you immobile. You'd better avoid it."
 	icon_state = "trap-shock"
 
 /obj/structure/trap/stun/trap_effect(mob/living/L)
-	L << "<span class='danger'><B>You are paralyzed from the intense shock!</B></span>"
+	L.electrocute_act(30, src, safety=1) // electrocute act does a message.
 	L.Weaken(5)
-	var/turf/Lturf = get_turf(L)
-	new /obj/effect/particle_effect/sparks/electricity(Lturf)
-	new /obj/effect/particle_effect/sparks(Lturf)
+	spark_system.start()
 
 
 /obj/structure/trap/fire
 	name = "flame trap"
 	desc = "A trap that will set you ablaze. You'd better avoid it."
 	icon_state = "trap-fire"
 
-
 /obj/structure/trap/fire/trap_effect(mob/living/L)
 	L << "<span class='danger'><B>Spontaneous combustion!</B></span>"
 	L.Weaken(1)
 	var/turf/Lturf = get_turf(L)
 	new /obj/effect/hotspot(Lturf)
-	new /obj/effect/particle_effect/sparks(Lturf)
+	spark_system.start()
 
 
 /obj/structure/trap/chill
 	name = "frost trap"
 	desc = "A trap that will chill you to the bone. You'd better avoid it."
 	icon_state = "trap-frost"
 
-
 /obj/structure/trap/chill/trap_effect(mob/living/L)
 	L << "<span class='danger'><B>You're frozen solid!</B></span>"
 	L.Weaken(1)
 	L.bodytemperature -= 300
-	new /obj/effect/particle_effect/sparks(get_turf(L))
+	spark_system.start()
```

Comment:
I think we have an actual frozen status effect now, it's normally used with freon